pkg/wait/conditions.go

```diff
@@ -149,6 +150,35 @@ func IsAppStatus(ctx context.Context, kubeClient *client.Client, appName string,
 	}
 }
 
+// IsAllAppStatus returns a WaitCondition that checks if all the apps provided currently have the provided expected status
+func IsAllAppStatus(ctx context.Context, kubeClient *client.Client, appNamespacedNames []types.NamespacedName, expectedStatus string) WaitCondition {
+	return func() (bool, error) {
+		var err error
+		isSuccess := true
+
+		for _, namespacedName := range appNamespacedNames {
+			app := &applicationv1alpha1.App{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      namespacedName.Name,
+					Namespace: namespacedName.Namespace,
+				},
+			}
+			if err = kubeClient.Client.Get(ctx, cr.ObjectKeyFromObject(app), app); err != nil {
+				isSuccess = false
+				continue
+			}
```

Comment:
nitpick: wouldn't be simpler like this?

```suggestion
			app := &applicationv1alpha1.App{}
			if err = kubeClient.Client.Get(ctx, namespacedName, app); err != nil {
				isSuccess = false
				continue
			}
```

Reply:
Very good point! Updated!